app/channel.go

```diff
@@ -209,6 +209,18 @@ func (a *App) CreateChannelWithUser(c *request.Context, channel *model.Channel,
 	message := model.NewWebSocketEvent(model.WebsocketEventChannelCreated, "", "", userID, nil)
 	message.Add("channel_id", channel.Id)
 	message.Add("team_id", channel.TeamId)
+
+	if category != nil {
+		category.Channels = append([]string{channel.Id}, category.Channels...)
+		var categories []*model.SidebarCategoryWithChannels
+		categories = append(categories, category)
```

Comment:
Any reason we are allocating a new slice here? Can we not pass `category` directly by doing `[]*model.SidebarCategoryWithChannels{category}`?